float32 left_angle
float32 left_radius
float32 right_angle
float32 right_radius
float32 rival_Dis
float32 rival_An
float32 border_Dis
float32 border_An




